tsfc/fem.py

```diff
@@ -319,9 +317,10 @@ def callback(key):
         table = ctx.tabulation_manager[key]
         if len(table.shape) == 1:
             # Cellwise constant
-            row = gem.Literal(table)
+            row = table
         else:
-            table = ctx.index_selector(lambda i: gem.Literal(table[i]), mt.restriction)
+            table = ctx.index_selector(lambda i: gem.partial_indexed(table, (i,)),
```

Comment:
Oh, I see, because tables aren't always tables any more.  How about updating `partial_indexed` to return a lower-rank literal if the `tensor` argument is a `Literal`?  Is that not too ugly?